gpflow/optimizers/scipy.py

```diff
@@ -102,13 +107,13 @@ def _callback(x):
         return _callback
 
     @staticmethod
-    def pack_tensors(tensors: Iterator[tf.Tensor]) -> tf.Tensor:
+    def pack_tensors(tensors: Iterable[tf.Tensor]) -> tf.Tensor:
```

Comment:
```suggestion
    def pack_tensors(tensors: Variables) -> tf.Tensor:
```
perhaps

Reply:
as well as similar cases below

Reply:
the function itself just needs something it can iterate over - that it'll be used for the variables is up to the caller, not the callee, no?

Reply:
yeah i get what you mean. i think either's fine. happy to leave as is